EpikCord/client/http_client.py

```diff
@@ -1,312 +1,233 @@
-
-            self.global_ratelimit.set()
-            bucket.event.set()
-
-            return await self.request(method, url, *args, **kwargs, attempt=attempt + 1)
-
-        if res.status >= HTTPCodes.SERVER_ERROR:
-            raise DiscordServerError5xx(body)
-
-        elif res.status == HTTPCodes.NOT_FOUND:
-            raise NotFound404(body)
-
-        elif res.status == HTTPCodes.FORBIDDEN:
-            raise Forbidden403(body)
-
-        elif not 300 > res.status >= 200:
-            raise DiscordAPIError(body)
-
-        async def dispose():
-            await asyncio.sleep(300)
-
-            with contextlib.suppress(KeyError):
-                del self.buckets[bucket_hash]
-
-        bucket.close_task.cancel()
-
-        bucket.close_task = asyncio.create_task(dispose())
-
-        return res
-
-    @staticmethod
-    async def log_request(res, body: Optional[dict] = None):
-        message = [
-            f"Sent a {res.request_info.method} to {res.url} "
-            f"and got a {res.status} response. ",
-            f"Content-Type: {res.headers['Content-Type']} ",
-        ]
-
-        if body:
-            message.append(f"Sent body: {body} ")
-
-        if h := dict(res.request_info.headers):
-            message.append(f"Sent headers: {h} ")
-
-        if h := dict(res.headers):
-            message.append(f"Received headers: {h} ")
-
-        try:
-            message.append(f"Received body: {await res.json()} ")
-
-        finally:
-            logger.debug("".join(message))
-
-    def base(
-        self,
-        method,
-        url,
-        *args,
-        to_discord: bool = True,
-        **kwargs,
-    ):
-        if to_discord:
-            return self.request(method, url, *args, **kwargs)
-        return self.session.request(method, url, *args, **kwargs)
-
-    get = partialmethod(base, "GET")
-    post = partialmethod(base, "POST")
-    put = partialmethod(base, "PUT")
-    patch = partialmethod(base, "PATCH")
-    delete = partialmethod(base, "DELETE")
-
-    async def get_gateway(self) -> discord_typings.GetGatewayData:
-        """
-        Get the gateway url.
-        Returns:
-            :class:`GetGatewayData`
-        """
-        res = await self.get("/gateway")
-        return await res.json()
-
-    async def get_gateway_bot(self) -> discord_typings.GetGatewayBotData:
-        """
-        Get information about the bot logging in, including the gateway url.
-        This is useful when sharding.
-
-        Returns:
-            :class:`GetGatewayBotData`
-        """
-        res = await self.get("/gateway/bot")
-        return await res.json()
-
-
-__all__ = ("HTTPClient",)
+                    bucket = Bucket(bucket_hash=response.headers["X-RateLimit-Bucket"])
+                    if bucket in self.buckets.values():
+                        self.buckets[f"{method}:{url}"] = list(self.buckets.values())[
+                            list(self.buckets.values()).index(bucket)
+                        ]
```

Comment:
This is very confusing code to follow through with. What exactly are you trying to do here?

Reply:
If the `Bucket` is already in other pre-existing routes, then the `Bucket` for this url is the already existing bucket.